src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -77,6 +70,20 @@ public UserDto update(UserDto newUserDto, User user) throws IllegalOperationExce
         return mapUserToDto(userRepository.saveAndFlush(user));
     }
 
+    public String updateUserPassword(UserChangePasswordDto userChangePasswordDto, User user) {
+        user.setPassword(bCryptPasswordEncoder.encode(userChangePasswordDto.getNewPassword()));
+        userRepository.saveAndFlush(user);
+
+        return getMessageSource("password.changed");
+    }
+
+    public String updateUserEmail(UserChangeEmailDto userChangeEmailDto, User user) {
+        user.setEmail(userChangeEmailDto.getNewEmail());
+        userRepository.saveAndFlush(user);
```

Comment:
As I see "email" it's unique parameter into the database what will happen if the user update email with existed one?

Reply:
will be done

Reply:
done